packages/features/bookings/Booker/components/hooks/useBookerTime.ts

```diff
@@ -0,0 +1,18 @@
+import { useBookerStore } from "@calcom/features/bookings/Booker/store";
+
+import { useTimePreferences } from "../../../lib/timePreferences";
+import { getBookerTimezone } from "../../utils/getBookerTimezone";
+
+export const useBookerTime = () => {
+  const [timezoneFromBookerStore] = useBookerStore((state) => [state.timezone]);
+  const { timezone: timezoneFromTimePreferences, timeFormat } = useTimePreferences();
+  const timezone = getBookerTimezone({
```

Comment:
Keeping in a separate utility in case it needs to be utilized by non component code.